### PR TITLE
Portal 1150: use localized web fonts

### PR DESCRIPTION
**Pull request by fearless-abbey**
This PR makes sure the UI is using the correct fonts by linking to localized web fonts. This PR also fixes the issue where the base font declarations were being applied at a global level, causing any downstream modifications to be overwritten.

To test that the fonts are working as expected:

1. Run `yarn storybook` and look at the rendered page
2. The page should default to the button component. Right click on the "Default Button:" label (this should be a `h5` element) and Inspect to open the dev tools.
3. Ensure locally installed fonts are disabled by clicking the three dots, "More tools," and "Rendering."
4. In the "Rendering" panel, click the checkbox next to the "Disable local fonts" option. With the panel still open, refresh the window to apply the changes.
Screenshot 2023-11-13 at 8 58 57 AM
Screenshot 2023-11-13 at 8 59 15 AM
Screenshot 2023-11-13 at 8 59 37 AM

5. Navigate to the "Elements" tab in the dev tools panel, and then to the "Computed" sub tab. Scroll to the very bottom to the "Rendered Fonts" section. You should see "Open Sans"
Screenshot 2023-11-14 at 11 52 46 AM

Doing this same process on the current live storybook instance will show the difference, that Helvetica is actually being rendered
Screenshot 2023-11-14 at 11 53 08 AM

### CODE REVIEW

**Comment by cfarmer-fearless** [comment]
I was able to follow the testing steps and verified the computed font is set to open sans.  However, I'm getting the following errors in my browser console: 
image

File: packages/cdc-react/src/scss/_fonts.scss [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  font-family: "Open Sans";
+  src:
+    url("/fonts/opensans-italic-webfont.woff2") format("woff2"),
+    url("/fonts/opensans-italic-webfont.woff") format("woff");
+  font-weight: $font-weight-regular;
+  font-style: italic;
+}
+
+//Medium
+//TO DO: Generate the .woff file for this font
```
Comment (cfarmer-fearless) [inline packages/cdc-react/src/scss/_fonts.scss:43]: What's the plan for the TODOs in this file?

**Comment by fearless-abbey** [comment]
> I was able to follow the testing steps and verified the computed font is set to open sans. However, I'm getting the following errors in my browser console: image
> 
> Let's talk about how these fonts will be consumed by the portal app.

Could this have something to do with rendering locally versus publishing? It's looking for the files in the fonts folder, which does compile as expected when running the `build` command. I'm a little out of my depth here however.

File: packages/cdc-react/src/scss/_fonts.scss (hunk 1 above)
Comment (fearless-abbey) [reply packages/cdc-react/src/scss/_fonts.scss:43]: Yes, good point (I added these so I wouldn't forget, and here we are ;) ). I'll look into this now. If I end up going down a rabbit hole, maybe we can team up, or make another ticket.

Comment (fearless-abbey) [reply packages/cdc-react/src/scss/_fonts.scss]: I worked some magic to convert the Medium and Medium Italic ttfs from Google Fonts to woff and woff2 formats, added them to the fonts folder, and removed the TO DOs.

**Comment by fearless-abbey** [comment]
> > I was able to follow the testing steps and verified the computed font is set to open sans. However, I'm getting the following errors in my browser console: image
> > Let's talk about how these fonts will be consumed by the portal app.
> 
> I looked into this further, and realized that "Disable Local Fonts" doesn't do what I thought it did. It does not disable locally installed fonts, instead it disables fonts implemented with the local() css function. I'm going back to the drawing board on this.

OK, I updated the storybook vite config to locate the font directory in the cdc-react library. This probably needs tweaking (not sure if this is the correct approach), but the console error regarding the font files has been resolved, and the Rendered Font is being identified as a Network Resource as opposed to Local file (which I missed initially).
Screenshot 2023-11-14 at 7 58 55 PM

**Comment by cfarmer-fearless** [comment]
> > > I was able to follow the testing steps and verified the computed font is set to open sans. However, I'm getting the following errors in my browser console: image
> > > Let's talk about how these fonts will be consumed by the portal app.
> > 
> > 
> > I looked into this further, and realized that "Disable Local Fonts" doesn't do what I thought it did. It does not disable locally installed fonts, instead it disables fonts implemented with the local() css function. I'm going back to the drawing board on this.
> 
> OK, I updated the storybook vite config to locate the font directory in the cdc-react library. This probably needs tweaking (not sure if this is the correct approach), but the console error regarding the font files has been resolved, and the Rendered Font is being identified as a Network Resource as opposed to Local file (which I missed initially). Screenshot 2023-11-14 at 7 58 55 PM

Let's look into this a bit more.  I'm no longer seeing the errors in storybook, but now I'm seeing the errors in the portal when linking the library locally.  It would be best if the font paths in cdc-react are able to resolve to the font files without any extra configuration on the consuming app side.  Will need some time to figure out how to do this though.

**Comment by fearless-abbey** [comment]
TO DO: Remove font family references in component css files.

**Comment by ian-sawyer** [comment]
I was able to get the fonts working in cdc-react and storybook. It loads them only if referenced in the CSS. 
Screenshot 2024-01-09 132625
Screenshot 2024-01-09 132709

**Comment by tbolt** [comment]
Working for me-
Screenshot 2024-01-30 at 1 29 44 PM

**Review by tbolt — APPROVED** [review]

**Review by sublettk — APPROVED** [review]
LGTM

**Pull request merged**